tomcat-logs/src/main/java/org/pentaho/tomcat/logvalve/FilteredAccessLogValve.java

```diff
@@ -35,10 +35,18 @@ public class FilteredAccessLogValve extends AccessLogValve {
   public void log( CharArrayWriter message ) {
     try ( CharArrayWriter caw = new CharArrayWriter() ) {
       // Mask the user password
-      caw.write( message.toString().replaceAll( "j_password=[^&^ ]*", "j_password=***" ) );
+      String tempString = message.toString();
+      if ( tempString.contains( "/pentaho/api/csrf" ) || tempString.contains( "/pentaho/api/repo/files/backup" ) ) {
+        tempString = tempString.replaceAll( "\\?userid[^&]+%26", "" );
+        tempString = tempString.replaceAll( "\\?userid[^&]+", "" );
+        tempString = tempString.replaceAll( "password[^&]+%26", "" );
+        tempString = tempString.replaceAll( "\\&password[^&]+", "" );
+      }
```

Comment:
IMO, this implementation should not have been merged as is.

Why does the filter only apply to these two URLs?
The parameter names used by `RequestParameterAuthenticationFilter` are configurable via the properties of the `requestParameterProcessingFilter` bean. Some customers may have these configured differently.